File: frontend/src/types/seller.ts
export interface Seller {
  id: number;
  companyName: string;
  description: string;
  logoUrl?: string;
  userName: string;
  userEmail: string;
  userId: number;
  isApproved: boolean;
  isSuspended: boolean;
  createdDate: string;
  totalProducts: number;
  totalRevenue: number;
}

export interface SellerDashboard {
  totalProducts: number;
  totalOrders: number;
  totalRevenue: number;
  recentOrders: SellerOrder[];
}

export interface SellerOrder {
  orderId: number;
  customerName: string;
  orderDate: string;
  status: string;
  sellerTotal: number;
  items: SellerOrderItem[];
}

export interface SellerOrderItem {
  productName: string;
  productImage: string;
  quantity: number;
  price: number;
  total: number;
}

File: backend/ECommerce.API/Models/Seller.cs
namespace ECommerce.API.Models;

public class Seller
{
    public int Id { get; set; }

    public string CompanyName { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string? LogoUrl { get; set; }

    public int UserId { get; set; }

    public User? User { get; set; }

    public bool IsApproved { get; set; } = true;

    public bool IsSuspended { get; set; } = false;

    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public ICollection<Product> Products { get; set; } = new List<Product>();
}
